Run 2 Workbooks (#151)
CodeYourFuture/immersive-go-course: website/content/workbooks/workbook-6.md

+++
title="CYF+02 Prep Workbook"
date="01 Jan 2024"    
versions=["1-1-0"]
weight=1
+++

# Prep

The CYF+ Immersive Engineering Programme is an intensive three month course. There's a lot to get through and you'll need to hit the ground running. To prepare for this exciting opportunity, you will:

> ## Linux

Reviewer: ```suggestion
## Computers

TODO: Add background reading on how computers work.

## Linux
```

- [ ] Play [the Bandit](https://overthewire.org/wargames/bandit/) - you must be able to complete up to level 20 (repeatedly)
- [ ] Take this [Unix 101 course](https://www.opsschool.org/unix_101.html) (and then try 102)
- [ ] Print out this [Linux-Cheat-Sheet](https://www.loggly.com/wp-content/uploads/2015/05/Linux-Cheat-Sheet-Sponsored-By-Loggly.pdf)

## Go

- [ ] Learn the basics of the Go programming language: [Get Started - The Go Programming Language](https://go.dev/learn/)
- [ ] Read this, you might find it useful for working through your first projects: [How to use the fmt package in Golang](https://www.educative.io/answers/how-to-use-the-fmt-package-in-golang)
- [ ] And optionally [Learn Go with tests](https://quii.gitbook.io/learn-go-with-tests/)

> ## Projects

Reviewer: ```suggestion
## Tooling
- [ ] [Set up and get to know your IDE](https://github.com/CodeYourFuture/immersive-go-course/tree/main/prep#set-up-and-get-to-know-your-ide)
- [ ] [Learnt how to navigate Go documentation](https://github.com/CodeYourFuture/immersive-go-course/tree/main/prep#learn-how-to-navigate-go-documentation)

## Projects
```

> Complete the prep and first _two_ projects from the [Immersive Go](https://github.com/CodeYourFuture/immersive-go-course) course

Reviewer: ```suggestion
Complete the [prep](https://github.com/CodeYourFuture/immersive-go-course/blob/main/prep/README.md) and first _five_ [projects](https://github.com/CodeYourFuture/immersive-go-course/blob/main/projects/README.md) from the [Immersive Go](https://github.com/CodeYourFuture/immersive-go-course) course
```
